components/faucet-claim.tsx

```diff
@@ -45,8 +46,15 @@ export const FaucetClaim = ({
   const activeChain = wallet?.getChain();
   const recaptchaRef = createRef() as React.RefObject<ReCAPTCHA>;
   const isMetaMask = wallet?.id === "io.metamask";
+  const { refetch: refetchWallet, data: balance } = useWalletBalance({
+    chain: mainnet,
+    address: wallet?.getAccount()?.address as string,
+    client,
+  });
+
   const minEthRequired = ENV.NODE_ENV === "production" ? 0.001 : 0.0001;
-  const isEligableToClaim = faucetStats && (faucetStats?.canClaim || faucetStats?.userMainnetBalance >= minEthRequired);
+  const isEligableToClaim =
+    faucetStats && (faucetStats?.canClaim || parseFloat(balance?.displayValue ?? "0") >= minEthRequired);
```

Comment:
We should use `&&` here here instead of `||` check. Because only if the user is both outside cooldown period and has the min. eth - they can claim

Reply:
this breaks because user is put into cooldown before claim is complete and this screen is showed 

![image](https://github.com/user-attachments/assets/f22fb427-d118-4983-b615-a59583d5b1d2)

Reply:
In that case, let's not show the message when the claim is in progress. That works?